src/Stratis.Bitcoin.Features.Interop/InteropPoller.cs

```diff
@@ -430,37 +434,31 @@ private async Task PollCirrusForBurnsAsync(BigInteger blockHeight)
                                 this.logger.Warn($"A dynamic fee for SRC20->ERC20 request '{receipt.TransactionHash}' could not be determined, using a fixed fee of {ConversionRequestFeeService.FallBackFee} CRS.");
                             }
 
-                            IFederation federation = this.network.Federations?.GetOnlyFederation();
-                            
-                            Script multisigScript = PayToFederationTemplate.Instance.GenerateScriptPubKey(federation.Id).PaymentScript;
-
-                            // Since we have no reliable way (yet) of extracting pricing data for all the potential tokens being transferred, there has to be a transaction output paying the multisig the conversion fee in order for the burn to be processed.
-                            // In future perhaps the fee could be taken out of the token value directly e.g. calculate a dollar fee and retain the equivalent SRC20 USDT. The distribution could then be done via an updated multisig contract instead.
-                            TxOut conversionFeeOutput = null;
-                            foreach (TxOut txOut in transaction.Outputs)
-                            {
-                                // For now, pay it directly to the multisig.
-                                if (txOut.ScriptPubKey == multisigScript)
-                                {
-                                    conversionFeeOutput = txOut;
-                                }
-                            }
+                            // Check if the fee and transfer amount is valid.
+                            byte[] valueBytes = src20burn.Value.ToByteArray();
+                            byte[] transferRequestAmountBytes = new byte[32];
+                            Array.Copy(valueBytes, 0, transferRequestAmountBytes, 0, valueBytes.Length);
 
-                            if (conversionFeeOutput == null)
+                            var transferRequestAmount = new BigInteger(transferRequestAmountBytes);
```

Comment:
Kevin please review the amounts carefully to ensure that my conversions to the various types are correct.